### ISSUE TITLE
Compilation fails when a scss (partial) files is updated

### ISSUE THREAD

**Issue by kossidts** [marko-js/marko#1819]
### Details

Compilation always fails as soon as I update (or just re-save) external scss files, I have added at the bottom of the app-layout/index.marko file i.e.
```html
<!DOCTYPE html>
<html lang="en">
...
</html>

style.scss {
   @use "/path/to/external/src/scss/styles.scss";
}
```

### Expected Behavior

The styles should compile and everything should work file, even when the `styles.scss` file is updated.

### Actual Behavior
While plain scss like e.g.
```html 
style.scss {
   .card {
        border: 1px solid var(--clr-gray-500);
        border-radius: 5px;

        h2 {
            font-size: 1.175rem;
        }
    }
}
```
works just fine, using `@use` or `@import` to include partials or any additional styles fails.
The first time the style block is added, everything seems to work fine. But as soon as I just re-save the `styles.scss` or any of its partials, the compilation fails with the error (see details section for the full error stack trace):

```
Compilation failed
✖ ｢wdm｣: Browser-dev:
  ERROR in ../components/app-layout/index.marko?virtual=./index.marko.scss (../components/app-layout/index.marko.webpack[javascript/auto]!=!../../node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[4].use[1]!../../node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[4].use[2]!../../node_modules/resolve-url-loader/index.js!../../../../node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[4].use[4]!../../node_modules/@marko/webpack/dist/loader/index.js!../components/app-layout/index.marko?virtual=./index.marko.scss)
  Module build failed (from ../../../../node_modules/sass-loader/dist/cjs.js):
  SassError: expected "{".
    ╷
  1 │ import { t as _t } from "marko/src/runtime/vdom/index.js";
    │                  ^
    ╵
    src/components/app-layout/index.marko 1:18  root stylesheet
 [...]
<w> [webpack.cache.PackFileCacheStrategy] Skipped not serializable cache item 'Compilation/modules|/Users/johnDoe/projects/project/src/frontend-marko/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[4].use[1]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[4].use[2]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/resolve-url-loader/index.js!/Users/johnDoe/projects/project/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[4].use[4]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/@marko/webpack/dist/loader/index.js!/Users/johnDoe/projects/project/src/frontend-marko/src/components/app-layout/index.marko?virtual=./index.marko.scss': No serializer registered for SassError
<w> while serializing webpack/lib/cache/PackFileCacheStrategy.PackContentItems -> webpack/lib/NormalModule -> webpack/lib/ModuleBuildError -> SassError
```

### Possible Fix

<details><summary>Additional Info</summary>

### Your Environment

- Environment name and version:
  - node.js 16.13.1
  - npm 8.1.2
  - marko 5.21.2
  - marko/webpack 9.2.2
  - marko/build 4.2.5
  - marko/compiler 5.21.6
  - marko/translator-default 5.21.2
- Operating System and version: macOs 12.3.1 w/ Apple M1 Pro

### Steps to Reproduce

1.  first...
2.
3.
4.

### Stack Trace

```
Compilation failed
✖ ｢wdm｣: Browser-dev:
  ERROR in ../components/app-layout/index.marko?virtual=./index.marko.scss (../components/app-layout/index.marko.webpack[javascript/auto]!=!../../node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[4].use[1]!../../node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[4].use[2]!../../node_modules/resolve-url-loader/index.js!../../../../node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[4].use[4]!../../node_modules/@marko/webpack/dist/loader/index.js!../components/app-layout/index.marko?virtual=./index.marko.scss)
  Module build failed (from ../../../../node_modules/sass-loader/dist/cjs.js):
  SassError: expected "{".
    ╷
  1 │ import { t as _t } from "marko/src/runtime/vdom/index.js";
    │                  ^
    ╵
    src/components/app-layout/index.marko 1:18  root stylesheet

  ERROR in ../components/app-layout/index.marko?virtual=./index.marko.scss (../components/app-layout/index.marko.scss!=!../../node_modules/@marko/webpack/dist/loader/index.js!../components/app-layout/index.marko?virtual=./index.marko.scss)
  Module build failed (from ../../node_modules/mini-css-extract-plugin/dist/loader.js):
  HookWebpackError: Module build failed (from ../../../../node_modules/sass-loader/dist/cjs.js):
  SassError: expected "{".
    ╷
  1 │ import { t as _t } from "marko/src/runtime/vdom/index.js";
    │                  ^
    ╵
    src/components/app-layout/index.marko 1:18  root stylesheet
      at tryRunOrWebpackError (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/HookWebpackError.js:88:9)
      at __webpack_require_module__ (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/Compilation.js:5055:12)
      at __webpack_require__ (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/Compilation.js:5012:18)
      at /Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/Compilation.js:5083:20
      at symbolIterator (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/neo-async/async.js:3485:9)
      at done (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/neo-async/async.js:3527:9)
      at Hook.eval [as callAsync] (eval at create (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
      at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/tapable/lib/Hook.js:18:14)
      at /Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/Compilation.js:4990:43
      at symbolIterator (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/neo-async/async.js:3482:9)
  -- inner error --
  Error: Module build failed (from ../../../../node_modules/sass-loader/dist/cjs.js):
  SassError: expected "{".
    ╷
  1 │ import { t as _t } from "marko/src/runtime/vdom/index.js";
    │                  ^
    ╵
    src/components/app-layout/index.marko 1:18  root stylesheet
      at Object.<anonymous> (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[4].use[1]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[4].use[2]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/resolve-url-loader/index.js!/Users/johnDoe/projects/project/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[4].use[4]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/@marko/webpack/dist/loader/index.js!/Users/johnDoe/projects/project/src/frontend-marko/src/components/app-layout/index.marko?virtual=./index.marko.scss:1:7)
      at /Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/javascript/JavascriptModulesPlugin.js:441:11
      at Hook.eval [as call] (eval at create (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:7:1)
      at Hook.CALL_DELEGATE [as _call] (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/tapable/lib/Hook.js:14:14)
      at /Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/Compilation.js:5057:39
      at tryRunOrWebpackError (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/HookWebpackError.js:83:7)
      at __webpack_require_module__ (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/Compilation.js:5055:12)
      at __webpack_require__ (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/Compilation.js:5012:18)
      at /Users/johnDoe/projects/project/src/frontend-marko/node_modules/webpack/lib/Compilation.js:5083:20
      at symbolIterator (/Users/johnDoe/projects/project/src/frontend-marko/node_modules/neo-async/async.js:3485:9)

  Generated code for /Users/johnDoe/projects/project/src/frontend-marko/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[4].use[1]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[4].use[2]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/resolve-url-loader/index.js!/Users/johnDoe/projects/project/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[4].use[4]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/@marko/webpack/dist/loader/index.js!/Users/johnDoe/projects/project/src/frontend-marko/src/components/app-layout/index.marko?virtual=./index.marko.scss
  1 | throw new Error("Module build failed (from ../../../../node_modules/sass-loader/dist/cjs.js):\nSassError: expected \"{\".\n  ╷\n1 │ import { t as _t } from \"marko/src/runtime/vdom/index.js\";\n  │                  ^\n  ╵\n  src/components/app-layout/index.marko 1:18  root stylesheet");

<w> [webpack.cache.PackFileCacheStrategy] Skipped not serializable cache item 'Compilation/modules|/Users/johnDoe/projects/project/src/frontend-marko/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[4].use[1]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[4].use[2]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/resolve-url-loader/index.js!/Users/johnDoe/projects/project/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[4].use[4]!/Users/johnDoe/projects/project/src/frontend-marko/node_modules/@marko/webpack/dist/loader/index.js!/Users/johnDoe/projects/project/src/frontend-marko/src/components/app-layout/index.marko?virtual=./index.marko.scss': No serializer registered for SassError
<w> while serializing webpack/lib/cache/PackFileCacheStrategy.PackContentItems -> webpack/lib/NormalModule -> webpack/lib/ModuleBuildError -> SassError
```

</details>

**Comment by DylanPiercey** [comment]
@kossidts could you create a reproduction repository with the issue and share that?

**Comment by kossidts** [comment]
Thank for the quick replay. I have created a github repository the for this purpose: https://github.com/kossidts/test-markojs

**Comment by DylanPiercey** [comment]
@kossidts really appreciate the report and reproduction here. It turns out the issue was related to how @marko/webpack handled virtual files. If those virtual files (like the `style.scss` block) had dependencies (like your `@use`) it could get wonky.

I just published a fix `@marko/webpack (9.2.3)`.

**Comment by kossidts** [comment]
Great job. Thanks.
